elastic/apm-server
# Fix legacy tracer initialization

## Motivation/summary

Change the approach to init `apm-server.instrumentation`: handle it in `beater.go` instead of relying on the config overrides logic

**Update**: additionally, enable the profile endpoint always, without checking the configuration. this wasn't working neither when using the new configuration from libbeat.  

## Checklist


I have considered changes for:

## How to test these changes

- Check that both `apm-server.instrumentation.*` and `instrumentation.*` settings work when reading config from a file or passing it with `-E` command args.

- Check that `instrumentation.profile.*` settings work as well.

- Check that `hosts` config works when explicitly set.

## Related issues

Fixes #3950
